core/drivers/stpmic1.c

```diff
@@ -663,15 +663,25 @@ int stpmic1_bo_voltage_cfg(const char *name, uint16_t millivolts,
 		return 1;
 
 	cfg->ctrl_reg = regul->control_reg;
-	cfg->value = voltage_index << LDO_BUCK_VOLTAGE_SHIFT;
+	cfg->min_value = min_index << LDO_BUCK_VOLTAGE_SHIFT;
 	cfg->mask = mask;
 
 	return 0;
 }
 
 int stpmic1_bo_voltage_unpg(struct stpmic1_bo_cfg *cfg)
 {
-	return stpmic1_register_update(cfg->ctrl_reg, cfg->value, cfg->mask);
+	uint8_t value = 0;
+
+	assert(cfg->ctrl_reg);
+
+	if (stpmic1_register_read(cfg->ctrl_reg, &value))
+		return -1;
+
+	value &= cfg->mask;
+	value = MIN(value, cfg->min_value);
```

Comment:
Should this be `MAX()`? If I'm not mistaken, `MIN()` won't do "stpmic1_bo_voltage_unpg() [...] is updated to not lower current voltage if above min voltage constraint."

Reply:
actually, should be:
```c
	if (stpmic1_register_read(cfg->ctrl_reg, &value))
		return -1;

	if (value & cfg->mask < cfg->min_value)
		return stpmic1_register_update(cfg->ctrl_reg, cfg->min_value, cfg->mask);

	return 0;